osf/models/registrations.py

```diff
@@ -647,11 +899,101 @@ def reject(self, user):
         self.approval.save()
 
     def add_status_log(self, user, action):
-        log = DraftRegistrationLog(action=action, user=user, draft=self)
+        params = {
+            'draft_registration': self._id,
+        },
+        log = DraftRegistrationLog(action=action, user=user, draft=self, params=params)
         log.save()
 
     def validate_metadata(self, *args, **kwargs):
         """
         Validates draft's metadata
         """
         return self.registration_schema.validate_metadata(*args, **kwargs)
+
+    def add_log(self, action, params, auth, save=True):
+        """
+        Tentative - probably need to combine with add_status_log
+        """
+        user = auth.user if auth else None
+
+        params['draft_registration'] = params.get('draft_registration') or self._id
+
+        log = DraftRegistrationLog(
+            action=action, user=user,
+            params=params, draft=self
+        )
+        log.save()
+        return log
+
+    def save(self, *args, **kwargs):
+        if 'old_subjects' in kwargs.keys():
+            kwargs.pop('old_subjects')
+        return super(DraftRegistration, self).save(*args, **kwargs)
+
+    def update(self, fields, auth=None, save=True):
+        """Update the draft registration with the given fields.
+
+        :param dict fields: Dictionary of field_name:value pairs.
+        :param Auth auth: Auth object for the user making the update.
+        :param bool save: Whether to save after updating the object.
+        """
+        if not fields:  # Bail out early if there are no fields to update
+            return False
+        for key, value in fields.items():
+            if key not in self.WRITABLE_WHITELIST:
+                continue
+            if key == 'title':
+                self.set_title(title=value, auth=auth, save=False)
+            elif key == 'description':
+                self.set_description(description=value, auth=auth, save=False)
+            elif key == 'category':
+                self.set_category(category=value, auth=auth, save=False)
+            elif key == 'node_license':
+                self.set_node_license(
+                    {
+                        'id': value.get('id'),
+                        'year': value.get('year'),
+                        'copyrightHolders': value.get('copyrightHolders') or value.get('copyright_holders', [])
+                    },
+                    auth,
+                    save=save
+                )
+        if save:
+            updated = self.get_dirty_fields()
+            self.save()
+
+        return updated
+
+
+class DraftRegistrationUserObjectPermission(UserObjectPermissionBase):
+    """
+    Direct Foreign Key Table for guardian - User models - we typically add object
+    perms directly to Django groups instead of users, so this will be used infrequently
+    """
+    content_object = models.ForeignKey(DraftRegistration, on_delete=models.CASCADE)
```

Comment:
Since draft registrations are getting contributors, go ahead and have them use guardian as well for storing permissions.  This makes sharing code with `ContributorMixin` super easy.